### PR TITLE
Filter rooms by type and availability

### LINKED ISSUE #3417: Filtering resources: Line below the input field

### PR DESCRIPTION
**Pull request by st3iny**
Fix #3417

~Requires https://github.com/nextcloud/cdav-library/pull/554~ via this PR

Implements filtering rooms by type and availability. The availability will be hidden from the search results if the user only searches for available rooms/resources.

I took all supported room types from https://github.com/nextcloud/calendar/issues/3184#issuecomment-872172487. Letting the user filter room types using a text input will reduce the usability. Maybe we can implement a way to define custom room types when we implement the frontend for room creation. Then we can populate the room type dropdown from the backend.

### CODE REVIEW

**Comment by ChristophWurst** [comment]
> I took all supported room types from [#3184 (comment)](https://github.com/nextcloud/calendar/issues/3184#issuecomment-872172487). Letting the user filter room types using a text input will reduce the usability. Maybe we can implement a way to define custom room types when we implement the frontend for room creation. Then we can populate the room type dropdown from the backend.

There is a list of predefined room types somewhere. @miaulalala will look it up and we'll make that our set of supported types. This change also has to be reflected in the rooms back-end app and possibly the APIs that transport room type (at least in the phpdoc).

File: src/components/Editor/Resources/ResourceListSearch.vue [hunk 1]
```diff
@@ -240,15 +251,34 @@ export default {
...
+		 * Format availability of a search result
+		 *
+		 * @param {boolean} isAvailable The availability state
+		 * @return {string} Human readable and localized availability
+		 */
+		formatAvailability(isAvailable) {
+			if (isAvailable) {
+				// TRANSLATORS room or resource is available due to not being booked yet
+				return this.$t('calendar', 'available')
+			} else {
```
Comment (tcitworld) [inline src/components/Editor/Resources/ResourceListSearch.vue:277]: You can return directly.

**Comment by tcitworld** [comment]
> There is a list of predefined room types somewhere.

https://github.com/nextcloud/server/blob/master/lib/public/Calendar/Room/IRoomMetadata.php

**Comment by ChristophWurst** [comment]
> Allowed values for this key include

What bothers me is the word *include*. We need a definitive list for the front-end because otherwise we don't know what options to offer to the user. But I just talked to @miaulalala and we think that those four options are already good enough. Google only has two types: meeting rooms and others.

**Comment by tcitworld** [comment]
The point of the values being namedspaced with the NC prefix allows this to do that without thinking about it too much anyway.

**Review by ChristophWurst — APPROVED** [review]
Tested and works. Is there anything left to do?

**Comment by st3iny** [comment]
> Is there anything left to do?

Nope. The draft status was only due to the missing cdav-lib dependency. I'll do one final test run.

File: src/components/Editor/Resources/ResourceListSearch.vue (hunk 1 above)
Comment (st3iny) [reply src/components/Editor/Resources/ResourceListSearch.vue:277]: Done

**Pull request merged**